### ISSUE TITLE
[CommEx] Tag Group To Category

### ISSUE THREAD

**Issue by ritayutingwang** [carousell/Product-Design-Sync#4]
# Goal
To let admin specify relevant categories (or keywords in the future) to their groups. This helps groups to be exposed to people who are interested in relevant topics and thus increase adoption of groups.
We thought about tagging the keywords or categories by Carousell staff. However, considering scalability and admin ownership, we decide to enable admins to tag categories themselves.

# Solution
In "Edit Group" page for Group Admin, there will be new section for related categories. Admin can choose up to 5 sub-categories (cannot choose cat/sub-cat with sub-cats under it) for the group. 

## Flow draft (iOS as example)
<img width="1270" alt="screen shot 2018-09-04 at 3 22 56 pm" src="https://user-images.githubusercontent.com/7291681/45016192-7ae05b80-b056-11e8-9961-9b0c9bee2dcb.png">

In this design, I apply the pattern Diona is using for multiple location in sell form. 

## Design vs Implementation
For V1, we will not implement suggested categories, unless we find a way to accurately suggest.

# Related feature 
## Groups in Browse Experience
We will use these data to surface relevant groups to users when they browse a category.
- If a group is tagged to sub-cat `Toys & Games / Video Gaming / Video Games`, the group is related to `Toys & Games` and `Toys & Games / Video Gaming` and `Toys & Games / Video Gaming / Video Games` and might be surfaced to users when browsing these categories or used to select relevant discussions to surface in browsing experience.

**Comment by keith-oh** [comment]
On suggestions, there has been academic studies around the levels of automation. You can refer to Table 1 in this article to decide which level we are at. Quite useful as we enter the era of AI and robotics.
http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.420.9319&rep=rep1&type=pdf

A minor detail that should be addressed from a Design System point of view. I'm looking at the line below Add Category. I see the line being used in other components in 2 different stroke values. Wonder what that line is supposed to communicate? Is it necessary?

Also curious about whether I can select a parent category. If not, then at the last level, do we still need the arrows, since it will not bring up a new page. Maybe we can explore a tree structure kind of picker?

**Comment by ritayutingwang** [comment]
@suganth-carousell @keith-oh thanks for the input!
I was actually discussing the parent category and child category thing with Huiyi. Originally I limit admins options to child categories for fairness (if group A is tagged to 5 parent categories while group B is tagged to 5 child categories, the exposures to potential audience is not fair) . But after discussing with Huiyi, we both agree to let admins choose whatever categories that make sense to them. We just have to come up with a system to surface the most relevant groups based on what parent category or child category they are in.

On @suganth-carousell  suggestion on *multi-picker*, I don't think we will build it. Reasons being that for now we don't have a way in the Carousell to choose multiple category. Unless some other journey will need this component, I feel this feature is fine with single picker because every group will probably use it once. They don't need to use it often. But say, if we want to let users choose multiple child categories in filter, then we might need to build one because this will often be used.

Similarly, we thought about helping them put in the right categories, but
1. Every group would probably use it once or twice in a year time. I don't think investing efforts building that will be the priority.
2. We don't have data to suggest categories for groups. There is nothing to learn from data especially for new group. And it is usually up to admins how they want to shape up the topic and direction for the group.

@keith-oh can you elaborate on the stroke value? Do you mean some of them is full width and some of them have 12px gap on the left side?

**Comment by keith-oh** [comment]
With the lines that have 12px gap of the left side, there's a thicker version and a thinner version. And there's also the line that indicate a text field, with gaps on both sides. So I just see a page with a lot of horizontal lines but not really seeing what each of them really represents.

**Comment by mailtokamalnayan** [comment]
I agree with @keith-oh regarding exploring the tree structure. As an admin I might want to tag my group to be as broad as possible because I don't want to limit the scope of discussions (use case might be the opposite as well where as an admin I want the group to be as niche as possible) so if I wanted to tag it to Toy & Games, how might I do that? 

Minor UI suggestion... "Add category" I think should be part of the Selected categories list without the section break.
